Task/#255 redesign statistics screen

OldMetalmind opened the pull request:
Closes #255 

**Description** 

Statistics screen has suffered a completely redesign since we have now access to more data related to current situation of Covid19.

**Changes** 

✅ New design of the statistics Screen
✅ Corrects minor problems with the StatusAPIService:
- Wrong endpoints
- Checks of .isOk

✅ dartfmt @ root
👎 No tests

OldMetalmind committed:
Use TextStyles class for text

OldMetalmind committed:
Add new Statistics screen

Fix TextStyle name

Add more TextStyles

Add color pale grey

Add all necessary widgets

Fix wrong endpoint ServerStatus & ifISOkay

Formatting & adds check for results isOk

Add base status API data

Add more call to Splash page

Fix host production link

Statistics Page & last update date

Add data connection

OldMetalmind committed:
Small fixes and removed "Hospitalized" card.

miguelpruivo commented on lib/bloc/base_bloc.dart:
```diff
@@ -57,7 +57,7 @@ class ResultStream<S, M> {
 enum StateStream { request, loading, success, fail }
 
-/// The ResultStream instance for requesting the case Stats
+/// [DEPRECATED] The ResultStream instance for requesting the case Stats
```
If the class is deprecated, I'd rather use `@deprecated` annotation here.

miguelpruivo commented on lib/extensions/string_extensions.dart:
```diff
@@ -0,0 +1,21 @@
+/// Capitalize the first letter of a String
+/// Source:
+/// https://stackoverflow.com/questions/29628989/how-to-capitalize-the-first-letter-of-a-string-in-dart
+extension StringExtension on String {
```
Lucky for us, Intl has a method already for this, there's no need for this extension at all. Just use `toBeginningOfSentenceCase()` which will return a first letter of a `String` in upper case.

miguelpruivo commented on lib/providers/stats_provider.dart:
```diff
@@ -14,7 +14,7 @@
 import 'package:flutter/material.dart';
 
-/// Creates a Provider for updating UI
+/// [DEPRECATED]  Creates a Provider for updating UI
```
Use `@deprecated` annotation instead.

miguelpruivo commented on lib/ui/screens/statistics/components/number_and_percentage.dart:
```diff
@@ -0,0 +1,59 @@
+    return Row(
+      children: <Widget>[
+        Container(
+          margin: EdgeInsets.all(4.0),
```
Add `const`.

miguelpruivo commented on lib/ui/screens/statistics/components/statistics_container.dart:
```diff
@@ -0,0 +1,41 @@
+  Widget build(BuildContext context) {
+    return Container(
+      child: child,
+      margin: EdgeInsets.only(bottom: 8.0),
```
Make it `const`

miguelpruivo commented on lib/ui/screens/statistics/components/statistics_divider.dart:
```diff
@@ -0,0 +1,29 @@
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: EdgeInsets.symmetric(vertical: 4.0),
```
Also `const` here :smile:

miguelpruivo commented on lib/ui/screens/statistics/components/statistics_squared_number.dart:
```diff
@@ -0,0 +1,35 @@
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.symmetric(
```
`const`

miguelpruivo commented on lib/ui/screens/statistics/components/statistics_vertical.dart:
```diff
@@ -0,0 +1,86 @@
+          Text(label, style: TextStyles.h3()),
+          NumberAndPercentageWidget(value: value, percentage: percentage),
+          Container(
+            margin: EdgeInsets.only(bottom: 8.0),
```
`const`

OldMetalmind committed:
Add MPRuivo recommended changes

- Use the intl as sentence method
- Add const to EdgeInsets when needed
- Use @deprecated in the classes
